### ISSUE TITLE
TypeError when trying to train or do inference on mac MPS

### ISSUE THREAD

**Issue by oliverwm1** [ai2cm/ace#300]
Users may want to do training or inference using GPUs on their Macs. We could update the `fme.get_device` function to default to MPS before fall backing to CPU. However when I tried that I got the following type error shown below.

Suggested solution: make sure all auxiliary variables like area and sigma coordinates are single precision when they are first computed.

```
train.py:425: in main
    trainer = Trainer(train_config)
train.py:96: in __init__
    self.train_data = get_data_loader(
../core/data_loading/getters.py:106: in get_data_loader
    dataset = get_dataset(params, requirements)
../core/data_loading/getters.py:75: in get_dataset
    dataset = XarrayDataset(params.dataset, requirements)
../core/data_loading/_xarray.py:156: in __init__
    self._sigma_coordinates = get_sigma_coordinates(first_dataset)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

ds = <xarray.Dataset>
Dimensions:  (time: 20, grid_yt: 16, grid_xt: 32)
Coordinates:
  * time     (time) int32 0 1 2 3 4 5 ...k_4     float64 5.0
    ak_5     float64 5.0
    bk_5     float64 6.0
    ak_6     float64 6.0
    bk_6     float64 7.0

    def get_sigma_coordinates(ds: xr.Dataset) -> SigmaCoordinates:
        """
        Get sigma coordinates from a dataset.
    
        Assumes that the dataset contains variables named `ak_N` and `bk_N` where
        `N` is the level number. The returned tensors are sorted by level number.
    
        Args:
            ds: Dataset to get sigma coordinates from.
        """
        ak_mapping = {
            int(v[3:]): torch.as_tensor(ds[v].values)
            for v in ds.variables
            if v.startswith("ak_")
        }
        bk_mapping = {
            int(v[3:]): torch.as_tensor(ds[v].values)
            for v in ds.variables
            if v.startswith("bk_")
        }
        ak_list = [ak_mapping[k] for k in sorted(ak_mapping.keys())]
        bk_list = [bk_mapping[k] for k in sorted(bk_mapping.keys())]
    
        if len(ak_list) == 0 or len(bk_list) == 0:
            raise ValueError("Dataset does not contain ak and bk sigma coordinates.")
    
        if len(ak_list) != len(bk_list):
            raise ValueError(
                "Expected same number of ak and bk coordinates, "
                f"got {len(ak_list)} and {len(bk_list)}."
            )
    
        return SigmaCoordinates(
>           ak=torch.as_tensor(ak_list, device=fme.get_device()),
            bk=torch.as_tensor(bk_list, device=fme.get_device()),
        )
E       TypeError: Cannot convert a MPS Tensor to float64 dtype as the MPS framework doesn't support float64. Please use float32 instead.

../core/data_loading/_xarray.py:68: TypeError
```

**Comment by oliverwm1** [comment]
Messed around with this a bit on my Mac. Got a weird pickling error when trying to train which seemed potentially tricky to fix. Getting inference working was straightforward, just had to cast vertical/horizontal coordinate tensors to float. Interestingly, the "unfold" function is not implemented on MPS yet, so the integer time-coarsening inference test failed.

In general, probably makes more sense to do our dev on CPU instead of on MPS, so not sure how to handle change to `get_device()`. Maybe could set it based on an env var we define like `FME_USE_MPS`.

**Comment by oliverwm1** [comment]
This was resolved by ai2cm/full-model#658. We can worry about how to allow user to configure the desire to use MPS later.
